### ISSUE TITLE
slow compilation

### ISSUE THREAD

**Issue by jwaldmann** [lehins/hip#48]
Hi. This library is very slow to compile? With ghc-8.10.2, compiling hip-1.5.6.0 from hackage takes more than 10 minutes  total (some modules take 5 min), on an otherwise reasonable machine ( i7 CPU         920  @ 2.67GHz). Is that due to repa?

**Comment by lehins** [comment]
Nah, it is not repa's fault. It is due to the way conversion of different image formats is implemented. 

I am in the somewhat slow, but gradual process of translating using massiv instead of repa. That issue that you mention here has already been fixed, but it will probably take another couple of months before I can make a release. Much documentation and tests need to be written, but if you feel adventurous you can give it a try use a version from master ;)

**Comment by jwaldmann** [comment]
Ah. - Perhaps submit current code as GHC test case ... ? (https://gitlab.haskell.org/ghc/ghc/-/issues?label_name[]=compiler%20perf )

**Comment by lehins** [comment]
It's a known issue, I don't have a ticket number for it however. Large modules with bunch of inlining/class instances have exponential time increase in compilation. Maybe this one, I am not sure: https://gitlab.haskell.org/ghc/ghc/-/issues/7428 
I wouldn't have my hopes up, ghc is pretty slow compiler in general, good thing at least it can still compile fast programs ;)
